File: Data Structures & Algorithms/sqrtx/submission-2.java
public class Solution {
    public int mySqrt(int x) {
        int l = 0, r = x;
        int res = 0;

        while (l <= r) {
            long m = l + (r - l) / 2;
            long sq = m * m;

            if (sq > x) {
                r = (int)m - 1;
            } else if (sq < x) {
                l = (int)m + 1;
                res = (int)m;
            } else {
                return (int)m;
            }
        }

        return res;
    }
}

File: Data Structures & Algorithms/minimum-number-of-moves-to-seat-everyone/submission-2.py
class Solution:
    def minMovesToSeat(self, seats: List[int], students: List[int]) -> int:
        max_index=max(max(seats),max(students))+1
        a=[0]*max_index
        b=[0]*max_index
        for seat in seats :
            a[seat]+=1
        for student in students :
            b[student]+=1
        remain =len(seats)
        i=0
        j=0
        res=0
        while remain :
            if a[i]==0:
                i+=1
            if b[j]==0:
                j+=1
            if a[i] and b[j] :
                res+=abs(i-j)
                a[i]-=1
                b[j]-=1
                remain-=1
        return res

        